cppapi/server/device.cpp

```diff
@@ -3229,7 +3229,43 @@ void DeviceImpl::add_attribute(Tango::Attr *new_attr)
 
 	if (new_attr->is_fwd() == true)
 	{
-		dev_attr->add_fwd_attribute(device_name,device_class,i,new_attr);
+		// If forwarded attribute is dynamically created and was constructed without specifying
+		// the root_attribute parameter then we have to get its __root_att property from tango DB
+		// prior to calling validate_fwd_att()
+		vector<AttrProperty> dev_prop;
+		Tango::Util *tg = Tango::Util::instance();
+		Tango::FwdAttr *fwd_attr = static_cast<Tango::FwdAttr *> (new_attr);
```

Comment:
Please, use `dynamic_cast` in this case. This cast specifically designed for inherited structures, see [reference](http://en.cppreference.com/w/cpp/language/dynamic_cast).